### PR TITLE
Allow cargo-generate 0.21.

### LINKED ISSUE #429: Templates do not build with the latest cargo-concordium cargo generate-generate 0.21.0
An attempt to build with the cargo generate cargo generate-generate 0.21.0 failed.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Fixes #429 

## Changes

Allow cargo generate 0.21. The changes from 0.20.* are not important https://github.com/cargo-generate/cargo-generate/releases/tag/v0.21.0

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Review by DOBEN — APPROVED** [review]

**Comment by abizjak** [comment]
@DOBEN I think you need to merge this when ready since I should not longer have write access.

**Pull request merged**
